staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -527,6 +527,19 @@ func (cnc *CloudNodeController) getProviderID(ctx context.Context, node *v1.Node
 		return node.Spec.ProviderID, nil
 	}
 
+	if instancesV2, ok := cnc.cloud.InstancesV2(); ok {
```

Comment:
Going back through previous discussions in https://github.com/kubernetes/kubernetes/pull/94894 and https://github.com/kubernetes/kubernetes/pull/92367, I think it was actually expected for providerID to be an empty string if InstanceV2 was not implemented. We don't really need providerID when we call InstanceMetadata on line 561 anyways. So I think here if InstanceV2 is enabled, we can just return empty providerID and nil error. What do you think?

Reply:
on Line 121 we make sure at least one type of **instances** is implemented
I agree that providerID should be an empty string if **InstancesV2** was not implemented, but in that case **Instances** must be implemented and we just run the rest of the logic for **Instances**
If InstanceV2 is enabled I agree we can just return an empty string, I just feel in that case probably we need to leave a note there

Reply:
Agreed, we should leave a comment regarding this